src/Controller/UserBadgesListBuilder.php

```diff
@@ -0,0 +1,153 @@
+  }
+
+  /**
+   * Provides listing of user badges.
+   *
+   * @param \Drupal\user\UserInterface $user
+   *   The user account.
+   *
+   * @return array
+   *   The block list as a renderable array.
+   */
+  public function badgesListing(UserInterface $user) {
+    $this->user = $user;
+    return $this->formBuilder->getForm($this);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getFormId() {
+    return 'badge_display_form';
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildForm(array $form, FormStateInterface $form_state) {
+    $entity_manager = \Drupal::entityManager();
+    $form['#attached']['library'][] = 'core/drupal.tableheader';
+    $form['#attributes']['class'][] = 'clearfix';
+    /** @var \Drupal\user\Entity\User $user */
+    $user = $this->user;
+
+    $form['badge_listing_table'] = array(
+      '#type' => 'table',
+      '#header' => array(t('Label'), t('Weight'), t('Select Weight')),
+      '#empty' => t('There are no badges yet'),
+      '#tableselect' => TRUE,
+      '#tabledrag' => array(
+        array(
+          'action' => 'order',
+          'relationship' => 'sibling',
+          'group' => 'user-badges-order-weight',
+        ),
+      ),
+    );
+    $badges = $user->get('field_user_badges')->getValue();
+    foreach ($badges as $badge_id) {
+      /** @var \Drupal\user_badges\Entity\Badge $badge */
+      $badge = $entity_manager->getStorage('badge')->load($badge_id['target_id']);
+      $form['badge_listing_table'][$badge->id()]['#attributes']['class'][] = 'draggable';
+      $form['badge_listing_table'][$badge->id()]['#weight'] = $badge->getBadgeWeight();
+
+      // Some table columns containing raw markup.
+      $form['badge_listing_table'][$badge->id()]['label'] = array(
+        '#plain_text' => $badge->label(),
+      );
+      $form['badge_listing_table'][$badge->id()]['id'] = array(
+        '#plain_text' => $badge->id(),
+      );
+
+      // TableDrag: Weight column element.
+      $form['badge_listing_table'][$badge->id()]['weight'] = array(
+        '#type' => 'weight',
+        '#title' => t('Weight for @title', array('@title' => $badge->label())),
+        '#title_display' => 'invisible',
+        '#default_value' => $badge->getBadgeWeight(),
+        '#attributes' => array('class' => array('user-badges-order-weight')),
+      );
+    }
+
+    $form['actions'] = array(
+      '#tree' => FALSE,
+      '#type' => 'actions',
+    );
+    $form['actions']['submit'] = array(
+      '#type' => 'submit',
+      '#value' => $this->t('Save Badges'),
+      '#button_type' => 'primary',
+    );
+
+    return $form;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function validateForm(array &$form, FormStateInterface $form_state) {
+    // No validation.
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function submitForm(array &$form, FormStateInterface $form_state) {
+    //$badges = $this->storage->loadMultiple(array_keys($form_state->getValue('badge_listing_table')));
```

Comment:
I am facing issue here.
when ever i **var_dump($form_state->getValue('badge_listing_table'));**
it gives me weird array structure. i.e it also adds table in form state as well. like given below.

```
array(3) {
  ["Array"]=>
  array(1) {
    ["weight"]=>
    string(3) "-10"
  }
  [2]=>
  array(1) {
    ["weight"]=>
    string(1) "2"
  }
  [1]=>
  array(1) {
    ["weight"]=>
    string(3) "-10"
  }
}
```